numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -1443,7 +1482,18 @@ static PyObject *npyiter_operands_get(NewNpyArrayIterObject *self)
                 "Iterator is invalid");
         return NULL;
     }
-
+    if ((self->managed & ITR_HAS_WRITEBACKIFCOPY) && 
+        !(self->managed & ITR_INSIDE)) {
+        /* 10-Nov-2017 v1.14 */
+        if (DEPRECATE(no_context_manager) < 0)
+            return NULL;
+    }
+    if (self->managed & ITR_EXITED) {
+        PyErr_SetString(PyExc_ValueError,
+                "Iterator has exited its context");
+        return NULL;
+    }
```

Comment:
Why is this desirable? How do `operands` interact with WRITEBACK_IF_COPY? Do they contain the temporary, or the target of the writeback?

Reply:
IIUC, you must call `PyArray_ResolveWritebackIfCopy` only once. I think operands contain the temporary copy, they operate on it, and then when the context has exited, the results are written back. Obviously they cannot be written back if the context has already exited.

Reply:
Not only that, but once they have been written back, the temporary scratch buffer is removed. Now the whole reason for the writeback semantics have been undone, for instance the operand dtype conflicts with the underlying base array's dtype. Reading from the operand will most likely retrieve the wrong value, or writing to it will corrupt the data.

Reply:
Would be ideal to recreate the scratch buffer in every `__enter__`. But fixing the deprecation is a higher priority IMHO.

Reply:
So to be clear, `operands` contains the temporary scratch buffer, not the original array?

Reply:
> I think operands contain the temporary copy

Reply:
operands are ``ndarrays`` that have the original array as a ``base`` attribute. Since they have the WRITEBACKIFCOPY flag, they hold another scratch buffer until ``PyArray_ResolveWritebackIfCopy`` is called